### PR TITLE
Homework3

### PR DESCRIPTION
**Pull request by komyagin**
Homework #3 is waiting for review

### CODE REVIEW

File: src/test/java/hw3/pages/AbstractPage.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package hw3.pages;
+
+import org.openqa.selenium.WebDriver;
+import hw3.pages.components.SideBarMenu;
+import hw3.pages.components.HeaderMenu;
+
+public class AbstractPage {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/pages/AbstractPage.java]: missed `abstract`

File: src/test/java/hw3/pages/JDITestingDifferentElementsPage.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+public class JDITestingDifferentElementsPage extends AbstractPage {
+
+    private LogsSection logSection;
+
+    private CheckboxRow checkboxRow;
+
+    private RadioButtonsRow radioRow;
+
+    private WebElement colorsDropdown;
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/pages/JDITestingDifferentElementsPage.java]: Why do you decide do not use `@FindBy` annotation?

File: src/test/java/hw3/pages/components/LogsSection.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    public boolean isDropdownLogDisplayed(String dropdownValue) {
+        boolean isDisplayed = false;
+        for (WebElement log : logs) {
+            if (log.getText().contains("Colors: value changed to " + dropdownValue)) {
+                isDisplayed = true;
+            }
+        }
+        return isDisplayed;
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/pages/components/LogsSection.java]: Looks like this is code duplication

File: src/test/java/hw3/pages/components/IFrame.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package hw3.pages.components;
+
+import hw3.pages.AbstractPageComposite;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class IFrame extends AbstractPageComposite {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/pages/components/IFrame.java]: You've called this class `IFrame`, but which **iframe** do you describe?

File: src/test/java/hw3/pages/components/RadioButtonsRow.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    public RadioButtonsRow(WebDriver driver) {
+        super(driver);
+    }
+
+    public void setRadio(String radioButtonName){
+        for (WebElement radioButtonElement : radioButtonElemetnts) {
+            if(!radioButtonElement.isSelected() & radioButtonElement.getText().equals(radioButtonName))
+                radioButtonElement.click();
+        }
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/pages/components/RadioButtonsRow.java]: This class is duplicating `CheckboxRow`. What is the purpose of this class?

File: src/test/java/hw3/pages/components/SideBarMenu.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        super(driver);
+    }
+
+    public boolean isSidebarElementsDisplayed(){
+        boolean isDisplayed = false;
+        for (WebElement element : sideBarMenuElements) {
+            isDisplayed = element.isDisplayed();
+        }
+        return isDisplayed;
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/pages/components/SideBarMenu.java]: Looks like this method is duplication

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Please resolve comments

File: src/test/java/hw3/pages/JDITestingDifferentElementsPage.java (hunk 2 above)
Comment (komyagin) [reply src/test/java/hw3/pages/JDITestingDifferentElementsPage.java]: These are instances of classes, not elements. I use @FindBy inside these classes

File: src/test/java/hw3/pages/AbstractPageComposite.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    public boolean isElementIsDisplayedOnSection(String elementName, List<WebElement> elements) {
+        boolean isDisplayed = false;
+        for (WebElement log : elements) {
+            if (log.getText().contains(": value changed to "  + elementName)) {
+                isDisplayed = true;
+            }
+        }
+        return isDisplayed;
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/pages/AbstractPageComposite.java:55]: Looks like duplication code a bit) You could generate log string somewhere else and pass it to the method as parameter

**Review by DmitryKhodakovsky — APPROVED** [review]
hw3 is accepted with comment. Could be merged to the master branch

**Pull request merged**